Adding some documentation on setting up LSP for neovim 0.8+

scottfines opened the pull request:
Should be a minor change to the Editors README on how to set up sourcekit-lsp for nevim 0.8 and above. The code segments are taken from my own tested configuration, along with as much notation as I could be reasonably confident in.

etcwilde reviewed (approved):
The indentation looks maybe a bit excessive (though personal opinion), the code snippet themselves look fine. Maybe add a few links to documentations on the tools so that folks can configure the server for themselves.

e.g https://neovim.io/doc/user/lsp.html#lsp-config, https://neovim.io/doc/user/lsp.html, etc...

scottfines commented:
I adjusted the indentation to line up with what I see in the rest of the doc (2 spaces), hopefully that will be more consistent. Also added a link to neovim's LSP configuration page for further information. There are some plugins that (theoretically) smooth this out, but since I haven't used them, I'm somewhat reluctant to even tangentially endorse them in the docs.

etcwilde commented on Editors/README.md:
````diff
@@ -143,6 +143,54 @@ As a test, open a Swift file, put the cursor on top of a symbol in normal mode a
+
+```lua
+require'lspconfig'.sourcekit.setup{
+        cmd = {'$TOOLCHAIN_PATH/usr/bin/sourcekit-lsp'}
````
One more to adjust the indentation on. :)

scottfines replied:
oops! my bad! fixed :)

etcwilde reviewed (approved):
Thanks for making those changes.

>  There are some plugins that (theoretically) smooth this out, but since I haven't used them, I'm somewhat reluctant to even tangentially endorse them in the docs.

I've been pretty happy with `vim-lsp`, though that's listed above. It works reasonably well with neovim too. I haven't tried any of the neovim/lua specific versions either.

etcwilde commented:
@swift-ci please smoke test

etcwilde commented:
@swift-ci please test

etcwilde commented:
Windows failure:
```
[swift]                                 Updating 'C:\Users\swift-ci\jenkins\workspace\sourcekit-lsp-PR-windows\swift'
[swift]                                 + git fetch --recurse-submodules=yes --tags
[swift]                                 b'From https://github.com/apple/swift'
[swift]                                 b' * branch                    HEAD       -> FETCH_HEAD'
[swift]                                 + git checkout main
[swift]                                 b"error: pathspec 'main' did not match any file(s) known to git"
```

etcwilde commented:
@swift-ci please test Windows

ahoppen commented:
@swift-ci Please test Windows

thecoldstone commented:
Just my 5 cents. After testing with different setups the most easiest one and less requiring does work perfectly with Neovim :)
